### PR TITLE
Feature/add permissions checking

### PR DESCRIPTION
**Pull request by mbenarie**

### CODE REVIEW

File: server/routes/departmentForms.js [hunk 1]
```diff
@@ -72,6 +83,10 @@ router.get("/form", co.wrap(function* (req, res) {
 router.post("/form", co.wrap(function* (req, res) {
     const form = req.body.form;
 
+    if (!permissionsUtils.isAdmin(req.userDetails)){
```
Comment (mbenarie) [inline server/routes/departmentForms.js]: this get should be accessible to all users

File: server/routes/departmentForms.js [hunk 2]
```diff
@@ -35,6 +34,10 @@ const saveDepartmentFrom = co.wrap(function* (departmentId, form) {
 router.get("/departments/:departmentId/forms", co.wrap(function* (req, res) {
     const departmentId = req.params.departmentId;
 
+    if (!permissionsUtils.isDepartmentManager(req.userDetails, departmentId)) {
```
Comment (mbenarie) [inline server/routes/departmentForms.js]: this get should be accessible to all users

File: server/routes/departments.js [hunk 3]
```diff
@@ -20,6 +20,10 @@ router.get('/departments/:departmentId', co.wrap(function* (req, res) {
 }));
 
 router.post('/departments', co.wrap(function* (req, res) {
+    if (!permissionsUtils.isAdmin(req.userDetails)){
```
Comment (mbenarie) [inline server/routes/departments.js]: this should be accessible to admins only

File: server/routes/departments.js [hunk 4]
```diff
@@ -45,6 +54,10 @@ router.put('/departments/:departmentId', co.wrap(function* (req, res) {
 }));
 
 router.delete('/departments/:departmentId', co.wrap(function* (req, res) {
+    if (!permissionsUtils.isAdmin(req.userDetails)){
```
Comment (mbenarie) [inline server/routes/departments.js]: this should be accessible to admins only

File: server/routes/permissions.js [hunk 5]
```diff
@@ -51,15 +49,17 @@ router.get('/permissions/admins', co.wrap(function* (req, res) {
 
 
 router.post('/permissions/admins', co.wrap(function* (req, res) {
-    const userId = req.userDetails.email;
-    const admin = yield Admin.find({userId: userId});
-    if (!admin) {
-        return res.status(403).json([{"error": "action is not allowed"}]);
+    if (!permissionsUtils.isAdmin(req.userDetails)) {
```
Comment (mbenarie) [inline server/routes/permissions.js]: this should be accessible to admins only

File: server/routes/departments.js (hunk 3 above)
Comment (talbarda) [reply server/routes/departments.js]: @may-ben-arie - it is, that's why you have this IF - if it's not the admin, it returns an error with 403 http code

File: server/routes/permissions.js (hunk 5 above)
Comment (talbarda) [reply server/routes/permissions.js]: @may-ben-arie - it is, that's why you have this IF - if it's not the admin, it returns an error with 403 http code

File: server/routes/departmentForms.js (hunk 2 above)
Comment (talbarda) [reply server/routes/departmentForms.js]: @may-ben-arie will be fixed

File: server/routes/departmentForms.js (hunk 1 above)
Comment (talbarda) [reply server/routes/departmentForms.js]: @may-ben-arie are you sure? isn't it adding the general form?? (Agreed on WhatsApp it's probably a mistake)

File: server/routes/departments.js (hunk 4 above)
Comment (talbarda) [reply server/routes/departments.js]: @may-ben-arie - it is, that's why you have this IF - if it's not the admin, it returns an error with 403 http code

**Pull request merged**